DOCS-10119-RTM-alerts (#157)
mulesoft/docs-monitoring: modules/ROOT/pages/basic-alerts.adoc

* *Resource name* or *Server name*
+
The application or server to apply the alert to
* *Condition*
+
The condition that triggers the alert
+
See <<alert-conditions>>.
* *Subject*
+
The subject of the alert email to send when the alert is triggered
+
// The subject can include variables.
// See <<alert-variables>>.
* *Message*
+
The body text for the email to send when the alert is triggered
// The message body can include variables.
// See <<alert-variables>>.
* *Recipients*
+
The recipients of the alert email
+
You can select up to 20 platform user IDs and enter up to 20 email addresses.
. Click *Create*.

image::basic-alerts-config.png[Configure a Basic Alert, 50%, 50%]

[[severity]]
=== Severity

The following levels of severity are available for basic alerts:

include::partial$include-alert-severity.adoc[leveloffset=+1]

[[alert-conditions]]
=== Alert Conditions

For basic alerts, alert conditions consist of a metric, an operator, and a period of time that the condition is met.

==== Metrics

The following metrics are available if your alert is created for an application:

Message count::
The total number of Mule messages that are processed successfully

Message error count::

> The total number of error messages. This includes 400- and 500- errors.

Reviewer: I do not think there is any relationship with 400 and 500 errors. @vailaya do we have a better definition for this metric? Can we just say its the number of mule messages which failed to be processed?

Reviewer 2: @ejaz-vel, @sarathecoubian  The mule runtime team described the error count as follows (https://www.mulesoft.org/jira/browse/SE-18424):

the message error count metric is incremented with every exception thrown in a flow.

Author: ```suggestion
The number of Mule messages that return an exception in a flow.
```

Reviewer 2: Sounds good

Message response time::
The response time of a message in milliseconds

> CPU utilization::

Reviewer: CPU/Memory/ThreadCount metrics are not available for all applications. Its only available for Cloudhub and RTF apps. For Hybrid Apps, we do not allow customers to create alerts on these 3 metrics, because it does not make sense

The percent of CPU used

Memory utilization::
The percent of memory used

Thread count::
The total number of threads used

> The following metrics are available if your alert is created for a server:

Reviewer: Not only for server. It can be a single server or a server group/cluster

Author: ```suggestion
The following metrics are available if your alert is created for a server, server group, or server cluster:
```

CPU utilization::

Memory utilization::

Thread count::

==== Operator

The _operator_ portion of the alert condition is the logic that is used against the metric you are measuring. The following operators are available:

* Above
* Above or equal
* Equal
* Below or equal
* Below

==== Time Period

The _time period_ defines how long the metric meets the criteria defined using the operator and chosen value. The following time periods are available:

* 5 minutes
* 10 minutes
* 15 minutes
* 20 minutes

> 

Reviewer: Should we also document that the customers should expect upto 15-20 minutes of lag before an alert is triggered?

Reviewer 2: @ejaz-vel @sarathecoubian The proper messaging, approved by the PM, is in JIRA ticket: https://www.mulesoft.org/jira/browse/DOCS-10278.

Author: Added in a different PR

. Click the *Alerts* tab.
. Click the *+ New Alerts* button.
. Click *Basic alert*.
. In the *Add basic alert* window, enter the information required to create your basic alert:

* *Name*
+
Unique and meaningful name for the alert
+
The *Name* is limited to 256 characters.
* *Severity level*
+
Severity of the alert: *Critical*, *Warning*, or *Info*
See <<severity>>.
+
* *Source type*
+
The source of the alert: applications or servers
+
* *Environment name*
+
The name of the environment where the resource is deployed
+
* *Resource name* or *Server name*
+
The application or server to apply the alert to
* *Condition*
+
The condition that triggers the alert
+
See <<alert-conditions>>.
* *Subject*
+
The subject of the alert email to send when the alert is triggered
+

> // The subject can include variables.
> // See <<alert-variables>>.

Author: ```suggestion
The subject can include variables.
See <<alert-variables>>.
```

* *Message*
+
The body text for the email to send when the alert is triggered

> // The message body can include variables.
> // See <<alert-variables>>.

Author: ```suggestion
The message body can include variables.
See <<alert-variables>>.
```

* *Recipients*
+
The recipients of the alert email
+
You can select up to 20 platform user IDs and enter up to 20 email addresses.
. Click *Create*.

image::basic-alerts-config.png[Configure a Basic Alert, 50%, 50%]

[[severity]]
=== Severity

The following levels of severity are available for basic alerts:

include::partial$include-alert-severity.adoc[leveloffset=+1]

[[alert-conditions]]
=== Alert Conditions

For basic alerts, alert conditions consist of a metric, an operator, and a period of time that the condition is met.

==== Metrics

The following metrics are available if your alert is created for an application:

Message count::
The total number of Mule messages that are processed successfully

Message error count::
The total number of error messages. This includes 400- and 500- errors.

Message response time::
The response time of a message in milliseconds

> CPU utilization::
> The percent of CPU used
> 
> Memory utilization::
> The percent of memory used
> 
> Thread count::
> The total number of threads used

Author: ```suggestion
CPU utilization::
The percent of CPU used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Memory utilization::
The percent of memory used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Thread count::
The total number of threads used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
```

> CPU utilization::
> The percent of CPU used
> 
> Memory utilization::
> The percent of memory used
> 
> Thread count::
> The total number of threads used
> 

Author: ```suggestion
```

The following metrics are available if your alert is created for a server, server group, or server cluster:

> CPU utilization::

Author: The percent of CPU used

> CPU utilization::

Author: ```suggestion
CPU utilization::
The percent of CPU used
```

> Memory utilization::

Author: ```suggestion
Memory utilization::
The percent of memory used
```

> Thread count::

Author: ```suggestion
Thread count::
The total number of threads used
```

See <<severity>>.
+
* *Source type*
+
The source of the alert: applications or servers
+
* *Environment name*
+
The name of the environment where the resource is deployed
+
* *Resource name* or *Server name*
+
The application or server to apply the alert to
* *Condition*
+
The condition that triggers the alert
+
See <<alert-conditions>>.
* *Subject*
+
The subject of the alert email to send when the alert is triggered
+
The subject can include variables.
See <<alert-variables>>.
* *Message*
+
The body text for the email to send when the alert is triggered
The message body can include variables.
See <<alert-variables>>.
* *Recipients*
+
The recipients of the alert email
+
You can select up to 20 platform user IDs and enter up to 20 email addresses.
. Click *Create*.

image::basic-alerts-config.png[Configure a Basic Alert, 50%, 50%]

[[severity]]
=== Severity

The following levels of severity are available for basic alerts:

include::partial$include-alert-severity.adoc[leveloffset=+1]

[[alert-conditions]]
=== Alert Conditions

For basic alerts, alert conditions consist of a metric, an operator, and a period of time that the condition is met.

> ==== Metrics

Author: ```suggestion
==== Application Metrics
```

. Click the *Alerts* tab.
. Click the *+ New Alerts* button.
. Click *Basic alert*.
. In the *Add basic alert* window, enter the information required to create your basic alert:

* *Name*
+
Unique and meaningful name for the alert
+
The *Name* is limited to 256 characters.
* *Severity level*
+
Severity of the alert: *Critical*, *Warning*, or *Info*
See <<severity>>.
+
* *Source type*
+

> The source of the alert: applications or servers

Reviewer 3: Look at/remove the "The"s?

> The source of the alert: applications or servers
> +
> * *Environment name*
> +
> The name of the environment where the resource is deployed
> +
> * *Resource name* or *Server name*
> +
> The application or server to apply the alert to

Author: ```suggestion
Source of the alert: applications or servers
+
* *Environment name*
+
Name of the environment where the resource is deployed
+
* *Resource name* or *Server name*
+
Application or server to apply the alert to
```

* *Condition*
+

> The condition that triggers the alert
> +
> See <<alert-conditions>>.
> * *Subject*
> +
> The subject of the alert email to send when the alert is triggered
> +
> The subject can include variables.
> See <<alert-variables>>.
> * *Message*
> +
> The body text for the email to send when the alert is triggered
> The message body can include variables.
> See <<alert-variables>>.
> * *Recipients*
> +
> The recipients of the alert email

Author: ```suggestion
Condition that triggers the alert
+
See <<alert-conditions>>.
* *Subject*
+
Subject of the alert email to send when the alert is triggered
+
The subject can include variables.
See <<alert-variables>>.
* *Message*
+
Body text for the email to send when the alert is triggered
The message body can include variables.
See <<alert-variables>>.
* *Recipients*
+
Recipients of the alert email
```

+
You can select up to 20 platform user IDs and enter up to 20 email addresses.
. Click *Create*.

image::basic-alerts-config.png[Configure a Basic Alert, 50%, 50%]

[[severity]]
=== Severity

The following levels of severity are available for basic alerts:

include::partial$include-alert-severity.adoc[leveloffset=+1]

[[alert-conditions]]
=== Alert Conditions

For basic alerts, alert conditions consist of a metric, an operator, and a period of time that the condition is met.

==== Application Metrics

The following metrics are available if your alert is created for an application:

Message count::

> The total number of Mule messages that are processed successfully

Reviewer 3: Only suggestion here is to remove the "The"s from these definitions.

Message error count::

> The number of Mule messages that return an exception in a flow.

Reviewer 3: ```suggestion
The number of Mule messages that return an exception in a flow
```

Message response time::

> The response time of a message in milliseconds

Reviewer 3: ```suggestion
The response time (in milliseconds) of a message
```

CPU utilization::

> The percent of CPU used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Reviewer 3: ```suggestion
The percent of CPU used.
+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
```

Memory utilization::

> The percent of memory used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Reviewer 3: ```suggestion
The percent of memory used. 
+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
```

Thread count::

> The total number of threads used. This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Reviewer 3: ```suggestion
The total number of threads used
+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
```

+
* *Environment name*
+
Name of the environment where the resource is deployed
+
* *Resource name* or *Server name*
+
Application or server to apply the alert to
* *Condition*
+
Condition that triggers the alert
+
See <<alert-conditions>>.
* *Subject*
+
Subject of the alert email to send when the alert is triggered
+
The subject can include variables.
See <<alert-variables>>.
* *Message*
+
Body text for the email to send when the alert is triggered
+
The message body can include variables.
See <<alert-variables>>.
* *Recipients*
+
Recipients of the alert email
+
You can select up to 20 platform user IDs and enter up to 20 email addresses.
. Click *Create*.

image::basic-alerts-config.png[Configure a Basic Alert, 50%, 50%]

[[severity]]
=== Severity

The following levels of severity are available for basic alerts:

include::partial$include-alert-severity.adoc[leveloffset=+1]

[[alert-conditions]]
=== Alert Conditions

For basic alerts, alert conditions consist of a metric, an operator, and a period of time that the condition is met.

==== Application Metrics

The following metrics are available if your alert is created for an application:

> Message count::
> The total number of Mule messages that are processed successfully
> 
> Message error count::
> The number of Mule messages that return an exception in a flow
> 
> Message response time::
> The response time (in milliseconds) of a message
> 
> CPU utilization::
> The percent of CPU used

Author: ```suggestion
Message count::
Total number of Mule messages that are processed successfully

Message error count::
Number of Mule messages that return an exception in a flow

Message response time::
Response time (in milliseconds) of a message

CPU utilization::
Percent of CPU used
```

+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Memory utilization::

> The percent of memory used
> +
> This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
> 
> Thread count::
> The total number of threads used
> +
> This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.
> 
> 
> ==== Server Metrics
> 
> The following metrics are available if your alert is created for a server, server group, or server cluster:
> 
> CPU utilization::
> The percent of CPU used
> 
> Memory utilization::
> The percent of memory used
> 
> Thread count::
> The total number of threads used

Author: ```suggestion
Percent of memory used
+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

Thread count::
Total number of threads used
+
This metric is available only for apps deployed to CloudHub or Anypoint Runtime Fabric.

==== Server Metrics

The following metrics are available if your alert is created for a server, server group, or server cluster:

CPU utilization::
Percent of CPU used

Memory utilization::
Percent of memory used

Thread count::
Total number of threads used
```
